resources/assets/js/forms.js

```diff
@@ -25,77 +17,99 @@ global.checkAll = (id, checked) => {
-
-  if (!fromDay || !fromTime || !toDay || !toTime) {
-    console.warn('cannot set input date because of missing field');
-    return;
+ready(() => {
+  /**
+   * @param {HTMLElement} element
+   */
+  const triggerChange = (element) => {
+    const changeEvent = new Event('change');
+    element.dispatchEvent(changeEvent);
   }
 
-  fromDay.value = formatDay(from);
-  triggerChange(fromDay);
-  fromTime.value = formatTime(from);
+  /**
+   * Sets the values of the input fields with the IDs to from/to:
+   * - date portion of from → start_day
+   * - time portion of from → start_time
+   * - date portion of to → end_day
+   * - time portion of to → end_time
+   *
+   * @param {Date} from
+   * @param {Date} to
+   */
+  const setInput = (from, to) => {
+    const fromDay = document.getElementById('start_day');
+    const fromTime = document.getElementById('start_time');
+    const toDay = document.getElementById('end_day');
+    const toTime = document.getElementById('end_time');
+
+    if (!fromDay || !fromTime || !toDay || !toTime) {
+      console.warn('cannot set input date because of missing field');
+      return;
+    }
 
-  toDay.value = formatDay(to);
-  triggerChange(toDay);
-  toTime.value = formatTime(to);
-};
+    fromDay.value = formatDay(from);
+    triggerChange(fromDay);
+    fromTime.value = formatTime(from);
 
-global.setDay = (days) => {
-  days = days || 0;
+    toDay.value = formatDay(to);
+    triggerChange(toDay);
+    toTime.value = formatTime(to);
+  };
 
-  const from = new Date();
-  from.setHours(0, 0, 0, 0);
+  /**
+   * @param {MouseEvent} event
+   */
+  const onClickDate = (event) => {
+    const days = Number(event.currentTarget.dataset.days);
 
-  // add days, Date handles the overflow
-  from.setDate(from.getDate() + days);
+    const from = new Date();
+    from.setHours(0, 0, 0, 0);
 
-  const to = new Date(from);
-  to.setHours(23, 59);
+    // add days, Date handles the overflow
+    from.setDate(from.getDate() + days);
 
-  setInput(from, to);
-};
+    const to = new Date(from);
+    to.setHours(23, 59);
 
-global.setHours = (hours) => {
-  hours = hours || 1;
+    setInput(from, to);
+  };
 
-  const from = new Date();
-  const to = new Date(from);
+  /**
+   * @param {MouseEvent} event
+   */
+  const onClickTime = (event) => {
+    const hours = Number(event.currentTarget.dataset.hours);
 
-  // convert hours to add to milliseconds (60 minutes * 60 seconds * 1000 for milliseconds)
-  const msToAdd = hours * 60 * 60 * 1000;
-  to.setTime(to.getTime() + msToAdd, 'h');
-  if (to < from) {
-    setInput(to, from);
-    return;
-  }
+    const from = new Date();
+    const to = new Date(from);
 
-  setInput(from, to);
-};
+    // add hours, Date handles the overflow
+    to.setHours(to.getHours() + hours);
+
+    if (to < from) {
+      setInput(to, from);
+    } else{
+      setInput(from, to);
+    }
+  };
+
+  document.querySelectorAll('.set-date').forEach((element) => {
```

Comment:
If we already have the data attributes, why not using them for selection as well? Something like this? So that we don't need to use the CSS class as functional marker.

```suggestion
  document.querySelectorAll('[data-set-days]').forEach((element) => {
```

Reply:
The plan was that the class choose what to set (time/date) and the data attribute the different units. so you could also easy use `data-minutes` etc in the future. should i still change it?